src/Microsoft.Data/PrimitiveColumn.cs

```diff
@@ -292,8 +293,6 @@ public override BaseColumn Clone(BaseColumn mapIndices = null, bool invertMapInd
                 Type dataType = mapIndices.DataType;
                 if (dataType != typeof(long) && dataType != typeof(int) && dataType != typeof(bool))
                     throw new ArgumentException(String.Format(Strings.MultipleMismatchedValueType, typeof(long), typeof(int), typeof(bool)), nameof(mapIndices));
-                if (mapIndices.Length > Length)
```

Comment:
Why is this being removed?

Reply:
For the new `Merge` algorithm in this patch. Because processing column-wise is more efficient, `Merge` with left, right and outer joins pass in a `mapIndices` larger than `Length`. So now, for ex: when the current column's length is say 5, but a right join produces a column of length 10, it is passed in a `mapIndices` that looks like [0, 1, 2, 3, 4, null, null, null, null, null]. Each null mapIndex appends a null in the resulting column